src/components/NewMovie/MovieForm.js

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { addMovie, newMovie, resetMovie } from "../../store/actionCreators";
+
+const MovieForm = (props) => {
+  const MovieTitleSelector = useSelector((state) => state.newMovie.title);
+  const MovieReleaseDateSelector = useSelector(
+    (state) => state.newMovie.release_date
+  );
+  const MoviePosterPathSelector = useSelector(
+    (state) => state.newMovie.poster_path
+  );
+  const MovieGenresSelector = useSelector((state) => state.newMovie.genres);
+  const MovieRuntimeSelector = useSelector((state) => state.newMovie.runtime);
+
+  const MovieOverviewSelector = useSelector((state) => state.newMovie.overview);
+
+  const dispatch = useDispatch();
+
+  return (
+    <div className="form-container">
+      <div className="form">
+        {MovieTitleSelector}
+        <button onClick={props.onCancel}>X</button>
+        <h1>Add Movie</h1>
+
+        <form onSubmit={() => dispatch(addMovie())} className="form__content">
+          <div className="form__contact-field-box">
+            <div>
+              <label>title</label>
+              <br />
+
+              <input
+                type="text"
+                className="form__input"
+                placeholder="Title"
+                value={MovieTitleSelector}
+                onChange={(e) =>
+                  dispatch(newMovie({ ...newMovie, title: e.target.value }))
```

Comment:
this component, as well as select below are uncontrolled component, since you're writing from e.target.value directly in redux. make it controlled.

it's true for all fields in this form. You need to write your data in store only when user clicks submit, not on every type. So, onChange should write in state of the component, and onSubmit will pop all this data altogether in Redux state

Reply:
This component has been replaced with Formik component

Reply:
let's get rid of it then :)

Reply:
resolved